<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Nunzio Knerr">
<meta name="author" content="Robert Godfree">
<meta name="dcterms.date" content="2024-11-29">

<title>pannotator examples</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body>

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#data-analysis-for-pannotator-paper" id="toc-data-analysis-for-pannotator-paper" class="nav-link active" data-scroll-target="#data-analysis-for-pannotator-paper">Data analysis for pannotator paper</a>
  <ul class="collapse">
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries">Load libraries</a></li>
  <li><a href="#read-species-richness-spinifex-cover-and-tree-health-.rds-files" id="toc-read-species-richness-spinifex-cover-and-tree-health-.rds-files" class="nav-link" data-scroll-target="#read-species-richness-spinifex-cover-and-tree-health-.rds-files">Read species richness, spinifex cover and tree health .rds files</a></li>
  <li><a href="#add-plot-and-frame-numbers-to-each-.rds-file" id="toc-add-plot-and-frame-numbers-to-each-.rds-file" class="nav-link" data-scroll-target="#add-plot-and-frame-numbers-to-each-.rds-file">Add plot and frame numbers to each .rds file</a></li>
  <li><a href="#map-the-location-of-data-points-in-each-file" id="toc-map-the-location-of-data-points-in-each-file" class="nav-link" data-scroll-target="#map-the-location-of-data-points-in-each-file">Map the location of data points in each file</a></li>
  </ul></li>
  <li><a href="#analysis-of-species-distribution-and-richness-data" id="toc-analysis-of-species-distribution-and-richness-data" class="nav-link" data-scroll-target="#analysis-of-species-distribution-and-richness-data">Analysis of species distribution and richness data</a>
  <ul class="collapse">
  <li><a href="#clean-up-the-species-data-df_species-for-mapping-and-analysis" id="toc-clean-up-the-species-data-df_species-for-mapping-and-analysis" class="nav-link" data-scroll-target="#clean-up-the-species-data-df_species-for-mapping-and-analysis">Clean up the species data (df_species) for mapping and analysis</a></li>
  <li><a href="#map-the-species-distributions-generate-image-for-figure-5a-in-paper" id="toc-map-the-species-distributions-generate-image-for-figure-5a-in-paper" class="nav-link" data-scroll-target="#map-the-species-distributions-generate-image-for-figure-5a-in-paper">Map the species distributions (generate image for Figure 5a in paper)</a></li>
  <li><a href="#make-a-25-m-buffer-around-each-plot-create-a-polygon-segment-it-and-calculate-species-diversity-in-each-segment" id="toc-make-a-25-m-buffer-around-each-plot-create-a-polygon-segment-it-and-calculate-species-diversity-in-each-segment" class="nav-link" data-scroll-target="#make-a-25-m-buffer-around-each-plot-create-a-polygon-segment-it-and-calculate-species-diversity-in-each-segment">Make a 25 m buffer around each plot, create a polygon, segment it and calculate species diversity in each segment</a></li>
  <li><a href="#make-buffer-around-each-point-and-create-polygons" id="toc-make-buffer-around-each-point-and-create-polygons" class="nav-link" data-scroll-target="#make-buffer-around-each-point-and-create-polygons">Make buffer around each point and create polygon(s)</a></li>
  <li><a href="#import-the-.kml-file-for-segmenting-the-polygons-and-cut-the-polygons" id="toc-import-the-.kml-file-for-segmenting-the-polygons-and-cut-the-polygons" class="nav-link" data-scroll-target="#import-the-.kml-file-for-segmenting-the-polygons-and-cut-the-polygons">Import the .kml file for segmenting the polygon(s) and cut the polygon(s)</a></li>
  <li><a href="#calculate-the-species-richness-in-each-of-the-new-polygons" id="toc-calculate-the-species-richness-in-each-of-the-new-polygons" class="nav-link" data-scroll-target="#calculate-the-species-richness-in-each-of-the-new-polygons">Calculate the species richness in each of the new polygons</a></li>
  <li><a href="#plot-the-centroids-of-each-polygon-with-species-richness-of-the-polygon-image-for-figure-5f" id="toc-plot-the-centroids-of-each-polygon-with-species-richness-of-the-polygon-image-for-figure-5f" class="nav-link" data-scroll-target="#plot-the-centroids-of-each-polygon-with-species-richness-of-the-polygon-image-for-figure-5f">Plot the centroids of each polygon with species richness of the polygon (image for Figure 5f)</a></li>
  </ul></li>
  <li><a href="#analysis-of-spinifex-data" id="toc-analysis-of-spinifex-data" class="nav-link" data-scroll-target="#analysis-of-spinifex-data">Analysis of spinifex data</a>
  <ul class="collapse">
  <li><a href="#check-the-data-and-correct-outliers" id="toc-check-the-data-and-correct-outliers" class="nav-link" data-scroll-target="#check-the-data-and-correct-outliers">Check the data and correct outliers</a></li>
  <li><a href="#map-the-total-cover-of-live-and-dead-cover-within-each-image-frame-images-for-figures-6c-and-6d" id="toc-map-the-total-cover-of-live-and-dead-cover-within-each-image-frame-images-for-figures-6c-and-6d" class="nav-link" data-scroll-target="#map-the-total-cover-of-live-and-dead-cover-within-each-image-frame-images-for-figures-6c-and-6d">Map the total cover of live and dead cover within each image frame (images for Figures 6c and 6d)</a></li>
  <li><a href="#map-the-cover-of-different-spinifex-size-classes-within-each-image-frame-image-for-figure-6e" id="toc-map-the-cover-of-different-spinifex-size-classes-within-each-image-frame-image-for-figure-6e" class="nav-link" data-scroll-target="#map-the-cover-of-different-spinifex-size-classes-within-each-image-frame-image-for-figure-6e">Map the cover of different spinifex size classes within each image frame (image for Figure 6e)</a></li>
  <li><a href="#re-name-spinifex-speciessize-data-for-size-class-analysis" id="toc-re-name-spinifex-speciessize-data-for-size-class-analysis" class="nav-link" data-scroll-target="#re-name-spinifex-speciessize-data-for-size-class-analysis">Re-name spinifex species/size data for size class analysis</a></li>
  <li><a href="#calculate-and-map-the-cover-and-distribution-of-50cm-and-fragmentary-old-1m-size-classes-image-for-figure-6e" id="toc-calculate-and-map-the-cover-and-distribution-of-50cm-and-fragmentary-old-1m-size-classes-image-for-figure-6e" class="nav-link" data-scroll-target="#calculate-and-map-the-cover-and-distribution-of-50cm-and-fragmentary-old-1m-size-classes-image-for-figure-6e">Calculate and map the cover and distribution of &lt; 50cm and fragmentary (old &gt; 1m) size classes (image for Figure 6e)</a></li>
  <li><a href="#create-boxplot-of-size-class-vs-pre-drought-and-post-drought-percentage-cover-with-kruskal-wallis-test-image-for-figure-6f" id="toc-create-boxplot-of-size-class-vs-pre-drought-and-post-drought-percentage-cover-with-kruskal-wallis-test-image-for-figure-6f" class="nav-link" data-scroll-target="#create-boxplot-of-size-class-vs-pre-drought-and-post-drought-percentage-cover-with-kruskal-wallis-test-image-for-figure-6f">Create boxplot of size class vs pre-drought and post-drought percentage cover with kruskal-wallis test (image for Figure 6f)</a></li>
  </ul></li>
  <li><a href="#analysis-of-desert-oak-crown-health-data" id="toc-analysis-of-desert-oak-crown-health-data" class="nav-link" data-scroll-target="#analysis-of-desert-oak-crown-health-data">Analysis of desert oak crown health data</a>
  <ul class="collapse">
  <li><a href="#import-the-lookup-csv-containing-size-classes-and-match-to-size-class-data-in-the-df_oak-object" id="toc-import-the-lookup-csv-containing-size-classes-and-match-to-size-class-data-in-the-df_oak-object" class="nav-link" data-scroll-target="#import-the-lookup-csv-containing-size-classes-and-match-to-size-class-data-in-the-df_oak-object">Import the lookup csv containing size classes and match to size class data in the df_oak object</a></li>
  <li><a href="#calculate-crown-health-score-and-compare-crown-health-across-size-classes-create-image-for-figure-7f" id="toc-calculate-crown-health-score-and-compare-crown-health-across-size-classes-create-image-for-figure-7f" class="nav-link" data-scroll-target="#calculate-crown-health-score-and-compare-crown-health-across-size-classes-create-image-for-figure-7f">Calculate crown health score and compare crown health across size classes (create image for Figure 7f)</a></li>
  <li><a href="#map-burnt-and-unburnt-tree-crowns-image-for-figure-7e" id="toc-map-burnt-and-unburnt-tree-crowns-image-for-figure-7e" class="nav-link" data-scroll-target="#map-burnt-and-unburnt-tree-crowns-image-for-figure-7e">Map burnt and unburnt tree crowns (image for Figure 7e)</a></li>
  <li><a href="#map-the-distribution-of-tree-size-classes-image-for-figure-7c" id="toc-map-the-distribution-of-tree-size-classes-image-for-figure-7c" class="nav-link" data-scroll-target="#map-the-distribution-of-tree-size-classes-image-for-figure-7c">Map the distribution of tree size classes (image for Figure 7c)</a></li>
  <li><a href="#plot-the-crown-health-of-mature-and-early-mature-size-classes-figure-7d" id="toc-plot-the-crown-health-of-mature-and-early-mature-size-classes-figure-7d" class="nav-link" data-scroll-target="#plot-the-crown-health-of-mature-and-early-mature-size-classes-figure-7d">Plot the crown health of mature and early mature size classes (Figure 7d)</a></li>
  </ul></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="pannotator_examples.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">pannotator examples</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p>Nunzio Knerr </p>
             <p>Robert Godfree </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 29, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="data-analysis-for-pannotator-paper" class="level1">
<h1>Data analysis for pannotator paper</h1>
<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load libraries</h2>
<p>This code loads in the libraries needed.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(mapview)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(rcompanion)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(sf)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">library</span>(webshot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-species-richness-spinifex-cover-and-tree-health-.rds-files" class="level2">
<h2 class="anchored" data-anchor-id="read-species-richness-spinifex-cover-and-tree-health-.rds-files">Read species richness, spinifex cover and tree health .rds files</h2>
<p>This code reads in the .Rds files containing species, spinifex cover and tree crown data.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># Read in the .Rds files containing species, spinifex cover and tree crown data</span></span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="co"># If you have exported an .Rds file from PannotatoR</span></span>
<span id="cb2-4"><a href="#cb2-4"></a>df_species <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"Data_files/df_species.Rds"</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co"># If you have exported an .csv file from PannotatoR</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>df_spinifex <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">"Data_files/df_spinifex.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb2-8"><a href="#cb2-8"></a></span>
<span id="cb2-9"><a href="#cb2-9"></a>df_oak <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"Data_files/oak_data_user1.Rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="add-plot-and-frame-numbers-to-each-.rds-file" class="level2">
<h2 class="anchored" data-anchor-id="add-plot-and-frame-numbers-to-each-.rds-file">Add plot and frame numbers to each .rds file</h2>
<p>Now we add the plot and information to each frame in the .rds file. This is useful code if you have multiple plots or subplots in a single .rds file. In this example we have only one plot (plot_06_new). A .csv file linking plot ID to frame or .kmz file ID is required.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># Now we add the plot and information to each frame in the .rds file. This is useful code if you have multiple plots or subplots in a single .rds file. In this example we have only one plot (plot_06_new). A .csv file linking plot ID to frame or .kmz file ID is required</span></span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co"># Import the csv file </span></span>
<span id="cb3-4"><a href="#cb3-4"></a>df_plot_number <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">"Data_files/Plot_names.csv"</span>) </span>
<span id="cb3-5"><a href="#cb3-5"></a></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co"># This function adds a plot_kmz column,fills it with a substring from the imagefile column, add a frame column with a substring from the imagefile column, and then match the plot number to the plot_kmz in the df_plot_number dataframe.</span></span>
<span id="cb3-7"><a href="#cb3-7"></a></span>
<span id="cb3-8"><a href="#cb3-8"></a>add_plot_kmz <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb3-9"><a href="#cb3-9"></a>  df<span class="sc">$</span>plot_kmz <span class="ot">&lt;-</span> <span class="fu">substr</span>(df<span class="sc">$</span>imagefile,<span class="dv">1</span>,<span class="dv">24</span>)</span>
<span id="cb3-10"><a href="#cb3-10"></a>  df<span class="sc">$</span>frame <span class="ot">&lt;-</span> <span class="fu">substring</span>(df<span class="sc">$</span>imagefile,<span class="dv">1</span>,<span class="dv">37</span> )</span>
<span id="cb3-11"><a href="#cb3-11"></a>  df<span class="sc">$</span>plot_name_new <span class="ot">&lt;-</span> df_plot_number<span class="sc">$</span>plot_name_new[<span class="fu">match</span>(df<span class="sc">$</span>plot_kmz, df_plot_number<span class="sc">$</span>plot)]</span>
<span id="cb3-12"><a href="#cb3-12"></a>  <span class="fu">return</span>(df)</span>
<span id="cb3-13"><a href="#cb3-13"></a>}</span>
<span id="cb3-14"><a href="#cb3-14"></a></span>
<span id="cb3-15"><a href="#cb3-15"></a>df_species <span class="ot">&lt;-</span> <span class="fu">add_plot_kmz</span>(df_species)</span>
<span id="cb3-16"><a href="#cb3-16"></a>df_spinifex <span class="ot">&lt;-</span> <span class="fu">add_plot_kmz</span>(df_spinifex)</span>
<span id="cb3-17"><a href="#cb3-17"></a>df_oak <span class="ot">&lt;-</span> <span class="fu">add_plot_kmz</span>(df_oak)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="map-the-location-of-data-points-in-each-file" class="level2">
<h2 class="anchored" data-anchor-id="map-the-location-of-data-points-in-each-file">Map the location of data points in each file</h2>
<p>This code maps the points for a quick look at the data.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># Using the geometry field</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>df_species <span class="ot">&lt;-</span> <span class="fu">st_as_sf</span>(df_species, <span class="at">wkt =</span> <span class="st">"geometry"</span>,<span class="at">crs =</span> <span class="dv">4326</span>) </span>
<span id="cb4-3"><a href="#cb4-3"></a>s1 <span class="ot">&lt;-</span> <span class="fu">mapview</span>(df_species)</span>
<span id="cb4-4"><a href="#cb4-4"></a></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="fu">mapshot</span>(s1, <span class="at">file =</span> <span class="st">"Data_files/species_map.png"</span>)</span>
<span id="cb4-6"><a href="#cb4-6"></a></span>
<span id="cb4-7"><a href="#cb4-7"></a>df_spinifex <span class="ot">&lt;-</span> <span class="fu">st_as_sf</span>(df_spinifex, <span class="at">wkt =</span> <span class="st">"geometry"</span>,<span class="at">crs =</span> <span class="dv">4326</span>) </span>
<span id="cb4-8"><a href="#cb4-8"></a>s2 <span class="ot">&lt;-</span> <span class="fu">mapview</span>(df_spinifex)</span>
<span id="cb4-9"><a href="#cb4-9"></a></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="fu">mapshot</span>(s2, <span class="at">file =</span> <span class="st">"Data_files/spinifex_map.png"</span>)</span>
<span id="cb4-11"><a href="#cb4-11"></a></span>
<span id="cb4-12"><a href="#cb4-12"></a>df_oak <span class="ot">&lt;-</span> <span class="fu">st_as_sf</span>(df_oak, <span class="at">wkt =</span> <span class="st">"geometry"</span>,<span class="at">crs =</span> <span class="dv">4326</span>) </span>
<span id="cb4-13"><a href="#cb4-13"></a>s3 <span class="ot">&lt;-</span> <span class="fu">mapview</span>(df_oak)</span>
<span id="cb4-14"><a href="#cb4-14"></a></span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="fu">mapshot</span>(s3, <span class="at">file =</span> <span class="st">"Data_files/desertoak_map.png"</span>)</span>
<span id="cb4-16"><a href="#cb4-16"></a></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb4-18"><a href="#cb4-18"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb4-19"><a href="#cb4-19"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb4-20"><a href="#cb4-20"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb4-21"><a href="#cb4-21"></a>  s1</span>
<span id="cb4-22"><a href="#cb4-22"></a>  s2</span>
<span id="cb4-23"><a href="#cb4-23"></a>  s3</span>
<span id="cb4-24"><a href="#cb4-24"></a>} <span class="cf">else</span> {</span>
<span id="cb4-25"><a href="#cb4-25"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb4-26"><a href="#cb4-26"></a>  s1</span>
<span id="cb4-27"><a href="#cb4-27"></a>  s2</span>
<span id="cb4-28"><a href="#cb4-28"></a>  s3</span>
<span id="cb4-29"><a href="#cb4-29"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<p><img src="Data_files/species_map.png" class="img-fluid quarto-figure quarto-figure-center"> <img src="Data_files/spinifex_map.png" class="img-fluid quarto-figure quarto-figure-center"> <img src="Data_files/desertoak_map.png" class="img-fluid quarto-figure quarto-figure-center"></p>
</div>
</section>
</section>
<section id="analysis-of-species-distribution-and-richness-data" class="level1">
<h1>Analysis of species distribution and richness data</h1>
<section id="clean-up-the-species-data-df_species-for-mapping-and-analysis" class="level2">
<h2 class="anchored" data-anchor-id="clean-up-the-species-data-df_species-for-mapping-and-analysis">Clean up the species data (df_species) for mapping and analysis</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># View(df_species)</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co"># Create a new species variable </span></span>
<span id="cb5-3"><a href="#cb5-3"></a>df_species<span class="sc">$</span>Species_new <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb5-4"><a href="#cb5-4"></a></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co"># If variable dd3 = -999, then add species (mulga information), otherwise keep the species listed in dd3. This annoatation was used to quantify mulga cover and also to add additional species #</span></span>
<span id="cb5-6"><a href="#cb5-6"></a>df_species<span class="sc">$</span>Species_new <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df_species<span class="sc">$</span>dd3 <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>, df_species<span class="sc">$</span>species, df_species<span class="sc">$</span>dd3)</span>
<span id="cb5-7"><a href="#cb5-7"></a></span>
<span id="cb5-8"><a href="#cb5-8"></a></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co"># For simplicity of data analysis we must now rename the species in the species_new variable </span></span>
<span id="cb5-10"><a href="#cb5-10"></a></span>
<span id="cb5-11"><a href="#cb5-11"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Acacia_aneura (mulga)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_aneura'</span></span>
<span id="cb5-12"><a href="#cb5-12"></a></span>
<span id="cb5-13"><a href="#cb5-13"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Acacia_ligulata (dune wattle)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_ligulata'</span></span>
<span id="cb5-14"><a href="#cb5-14"></a></span>
<span id="cb5-15"><a href="#cb5-15"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Acacia_minyura (desert mulga)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_minyura'</span></span>
<span id="cb5-16"><a href="#cb5-16"></a></span>
<span id="cb5-17"><a href="#cb5-17"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Allocasuarina_decaisneana (oak)'</span>] <span class="ot">&lt;-</span><span class="st">'Allocasuarina_decaisneana'</span></span>
<span id="cb5-18"><a href="#cb5-18"></a></span>
<span id="cb5-19"><a href="#cb5-19"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Burnt Mulga Mature (3-6m)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_aneura'</span></span>
<span id="cb5-20"><a href="#cb5-20"></a></span>
<span id="cb5-21"><a href="#cb5-21"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Burnt Mulga Juvenile (1-3m)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_aneura'</span></span>
<span id="cb5-22"><a href="#cb5-22"></a></span>
<span id="cb5-23"><a href="#cb5-23"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Burnt Mulga Overmature (3-6m)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_aneura'</span></span>
<span id="cb5-24"><a href="#cb5-24"></a></span>
<span id="cb5-25"><a href="#cb5-25"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Codonocarpus_cotinifolius (poplar)'</span>] <span class="ot">&lt;-</span><span class="st">'Codonocarpus_cotinifolius'</span></span>
<span id="cb5-26"><a href="#cb5-26"></a></span>
<span id="cb5-27"><a href="#cb5-27"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Grevillea_eriostachya (honey grevillea)'</span>] <span class="ot">&lt;-</span><span class="st">'Grevillea_eriostachya'</span></span>
<span id="cb5-28"><a href="#cb5-28"></a></span>
<span id="cb5-29"><a href="#cb5-29"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Grevillea_stenobotrys (sandhill grevillea)'</span>] <span class="ot">&lt;-</span><span class="st">'Grevillea_stenobotrys'</span></span>
<span id="cb5-30"><a href="#cb5-30"></a></span>
<span id="cb5-31"><a href="#cb5-31"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Gyrostemon_ramulosus (camel poison)'</span>] <span class="ot">&lt;-</span><span class="st">'Gyrostemon_ramulosus'</span></span>
<span id="cb5-32"><a href="#cb5-32"></a></span>
<span id="cb5-33"><a href="#cb5-33"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Mulga Juvenile (1-3m)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_aneura'</span></span>
<span id="cb5-34"><a href="#cb5-34"></a></span>
<span id="cb5-35"><a href="#cb5-35"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Mulga Mature (3-6m)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_aneura'</span></span>
<span id="cb5-36"><a href="#cb5-36"></a></span>
<span id="cb5-37"><a href="#cb5-37"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Mulga Seedling (&lt;1m)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_aneura'</span></span>
<span id="cb5-38"><a href="#cb5-38"></a></span>
<span id="cb5-39"><a href="#cb5-39"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Santalum_lanceolatum (sandalwood)'</span>] <span class="ot">&lt;-</span><span class="st">'Santalum_lanceolatum'</span></span>
<span id="cb5-40"><a href="#cb5-40"></a></span>
<span id="cb5-41"><a href="#cb5-41"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Themeda_australis (kangaroo grass)'</span>] <span class="ot">&lt;-</span><span class="st">'Crotalaria_cunninghamii'</span></span>
<span id="cb5-42"><a href="#cb5-42"></a></span>
<span id="cb5-43"><a href="#cb5-43"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Themeda_avenaceae (oat grass)'</span>] <span class="ot">&lt;-</span><span class="st">'Crotalaria_cunninghamii'</span></span>
<span id="cb5-44"><a href="#cb5-44"></a></span>
<span id="cb5-45"><a href="#cb5-45"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Acacia_kempeana (granite wattle)'</span>] <span class="ot">&lt;-</span><span class="st">'Acacia_kempeana'</span></span>
<span id="cb5-46"><a href="#cb5-46"></a></span>
<span id="cb5-47"><a href="#cb5-47"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Corymbia_opaca (desert bloodwood)'</span>] <span class="ot">&lt;-</span><span class="st">'Corymbia_opaca'</span></span>
<span id="cb5-48"><a href="#cb5-48"></a></span>
<span id="cb5-49"><a href="#cb5-49"></a>df_species<span class="sc">$</span>Species_new[df_species<span class="sc">$</span>Species_new <span class="sc">==</span> <span class="st">'Eucalyptus_gamophylla (blue mallee)'</span>] <span class="ot">&lt;-</span><span class="st">'Eucalyptus_gamophylla'</span></span>
<span id="cb5-50"><a href="#cb5-50"></a></span>
<span id="cb5-51"><a href="#cb5-51"></a><span class="co"># We now Clean up a few more cases. We don't want any data points that contain no species data, unknown species, or species with unclear taxonomy (e.g., Hakea spp.) </span></span>
<span id="cb5-52"><a href="#cb5-52"></a></span>
<span id="cb5-53"><a href="#cb5-53"></a>df_species <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_species, Species_new <span class="sc">!=</span> <span class="st">"No Mulga present"</span>)</span>
<span id="cb5-54"><a href="#cb5-54"></a>df_species <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_species, Species_new <span class="sc">!=</span> <span class="st">"Re_Check_Unknown_Shrub"</span>)</span>
<span id="cb5-55"><a href="#cb5-55"></a>df_species <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_species, Species_new <span class="sc">!=</span> <span class="st">"Hakea_sp"</span>)</span>
<span id="cb5-56"><a href="#cb5-56"></a>df_species <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_species, Species_new <span class="sc">!=</span> <span class="st">"Acacia_sp"</span>)</span>
<span id="cb5-57"><a href="#cb5-57"></a>df_species <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_species, Species_new <span class="sc">!=</span> <span class="st">"Gyrostemon_sp."</span>)</span>
<span id="cb5-58"><a href="#cb5-58"></a></span>
<span id="cb5-59"><a href="#cb5-59"></a><span class="co"># Remove species and location duplicates </span></span>
<span id="cb5-60"><a href="#cb5-60"></a>duplicated_rows <span class="ot">&lt;-</span> <span class="fu">duplicated</span>(df_species[<span class="fu">c</span>(<span class="st">"Species_new"</span>, <span class="st">"geometry"</span>)])</span>
<span id="cb5-61"><a href="#cb5-61"></a>df_species<span class="ot">&lt;-</span> df_species[<span class="sc">!</span>duplicated_rows, ]</span>
<span id="cb5-62"><a href="#cb5-62"></a></span>
<span id="cb5-63"><a href="#cb5-63"></a><span class="co"># We must now jitter the geometries so that we can see the data easier in mapview. Many of the locations are very close together </span></span>
<span id="cb5-64"><a href="#cb5-64"></a>df_species_jittered <span class="ot">&lt;-</span> <span class="fu">st_jitter</span>(df_species, <span class="at">amount =</span> <span class="fl">0.00001</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="map-the-species-distributions-generate-image-for-figure-5a-in-paper" class="level2">
<h2 class="anchored" data-anchor-id="map-the-species-distributions-generate-image-for-figure-5a-in-paper">Map the species distributions (generate image for Figure 5a in paper)</h2>
<p>This is useful code for plotting species distributions in mapview, including setting colour schemes within a user-specified species order.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co"># This code generates Figure 5A </span></span>
<span id="cb6-2"><a href="#cb6-2"></a></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co"># Confirm the data </span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co">#mapview(df_species_jittered, zcol = "Species_new", cex = 8, alpha = 0.9) </span></span>
<span id="cb6-5"><a href="#cb6-5"></a></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co"># Here I reorder the species from sand dune crest species to swale species (approximately) </span></span>
<span id="cb6-7"><a href="#cb6-7"></a></span>
<span id="cb6-8"><a href="#cb6-8"></a>df_species_jittered<span class="sc">$</span>speciesReorder <span class="ot">&lt;-</span> <span class="fu">factor</span>(df_species_jittered<span class="sc">$</span>Species_new, <span class="fu">c</span>(<span class="st">'Crotalaria_cunninghamii'</span>,<span class="st">'Gyrostemon_ramulosus'</span>, <span class="st">'Acacia_ligulata'</span>,   <span class="st">'Aluta_maisonneuvei'</span>, <span class="st">'Newcastelia_spodiotricha'</span>, <span class="st">'Gyrostemon_tepperi'</span>,<span class="st">'Codonocarpus_cotinifolius'</span>, <span class="st">'Grevillea_eriostachya'</span>, <span class="st">'Allocasuarina_decaisneana'</span>, <span class="st">'Exocarpus_sparteus'</span>,  <span class="st">'Eremophila_sp'</span>, <span class="st">'Eucalyptus_Mallee_sp.'</span>, <span class="st">'Eucalyptus_gamophylla'</span>, <span class="st">'Corymbia_opaca'</span>, <span class="st">'Acacia_kempeana'</span> ,<span class="st">'Acacia_aneura'</span>))</span>
<span id="cb6-9"><a href="#cb6-9"></a></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co"># Here I colour species groups according to their typical ecological position; e.g., sand dune crest species in red, swale species in white</span></span>
<span id="cb6-11"><a href="#cb6-11"></a>p <span class="ot">&lt;-</span> <span class="fu">mapview</span>(df_species_jittered, <span class="at">zcol =</span> <span class="st">"speciesReorder"</span>, <span class="at">cex =</span> <span class="dv">5</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>, <span class="at">layer.name =</span> <span class="st">"Species"</span>, <span class="at">col.regions =</span> <span class="fu">c</span>(<span class="st">"red4"</span>,<span class="st">"red4"</span>, <span class="st">'red4'</span>, <span class="st">"orange"</span>, <span class="st">'yellow3'</span>,<span class="st">'yellow3'</span>, <span class="st">'yellow2'</span>,<span class="st">'yellow2'</span>,<span class="st">'green3'</span>,<span class="st">'green2'</span>,<span class="st">'green4'</span>, <span class="st">'deepskyblue2'</span>, <span class="st">'deepskyblue2'</span>,<span class="st">'purple'</span>,<span class="st">'white'</span>,<span class="st">'white'</span>), </span>
<span id="cb6-12"><a href="#cb6-12"></a>        <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-13"><a href="#cb6-13"></a></span>
<span id="cb6-14"><a href="#cb6-14"></a><span class="fu">mapshot</span>(p, <span class="at">file =</span> <span class="st">"Data_files/Fig_5a.png"</span>)</span>
<span id="cb6-15"><a href="#cb6-15"></a></span>
<span id="cb6-16"><a href="#cb6-16"></a></span>
<span id="cb6-17"><a href="#cb6-17"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb6-18"><a href="#cb6-18"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb6-19"><a href="#cb6-19"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb6-20"><a href="#cb6-20"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb6-21"><a href="#cb6-21"></a>  p</span>
<span id="cb6-22"><a href="#cb6-22"></a>} <span class="cf">else</span> {</span>
<span id="cb6-23"><a href="#cb6-23"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb6-24"><a href="#cb6-24"></a>  p</span>
<span id="cb6-25"><a href="#cb6-25"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data_files/Fig_5a.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</div>
</section>
<section id="make-a-25-m-buffer-around-each-plot-create-a-polygon-segment-it-and-calculate-species-diversity-in-each-segment" class="level2">
<h2 class="anchored" data-anchor-id="make-a-25-m-buffer-around-each-plot-create-a-polygon-segment-it-and-calculate-species-diversity-in-each-segment">Make a 25 m buffer around each plot, create a polygon, segment it and calculate species diversity in each segment</h2>
<p>This is useful code to add a buffer around each data point in the dataframe, combined these together if overlapping into a series of polygons. The intent is to then split up the polygons according to another spatial dataframe, in this case a .kml containing lines that divide the west side of the transect into 50 m segments with approximately homogeneous vegetation.</p>
</section>
<section id="make-buffer-around-each-point-and-create-polygons" class="level2">
<h2 class="anchored" data-anchor-id="make-buffer-around-each-point-and-create-polygons">Make buffer around each point and create polygon(s)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># Add a spatial buffer of 25m around each point</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>buffer <span class="ot">&lt;-</span> <span class="fu">st_buffer</span>(df_species_jittered, <span class="at">dist =</span> <span class="dv">50</span>)</span>
<span id="cb7-3"><a href="#cb7-3"></a></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co"># Combine the resulting buffers into a single polygon</span></span>
<span id="cb7-5"><a href="#cb7-5"></a>polygon <span class="ot">&lt;-</span> <span class="fu">st_union</span>(buffer)</span>
<span id="cb7-6"><a href="#cb7-6"></a></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co"># mapview(polygon) # we can see that the buffers have been merged into a series of polygons.</span></span>
<span id="cb7-8"><a href="#cb7-8"></a></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co"># Export the resulting polygon as a .kml file</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="fu">st_write</span>(polygon, <span class="st">"Data_files/buffer_plot_polygon.kml"</span>, <span class="at">append=</span> <span class="cn">FALSE</span>, <span class="at">driver =</span> <span class="st">"KML"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="import-the-.kml-file-for-segmenting-the-polygons-and-cut-the-polygons" class="level2">
<h2 class="anchored" data-anchor-id="import-the-.kml-file-for-segmenting-the-polygons-and-cut-the-polygons">Import the .kml file for segmenting the polygon(s) and cut the polygon(s)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># This is.kml file that contains lines that cut the westerly side of the study transect into a series of segments roughly 50 m apart. Each segment was oriented to contain homogenous vegetation. It was drawn in Google Earth. </span></span>
<span id="cb8-2"><a href="#cb8-2"></a></span>
<span id="cb8-3"><a href="#cb8-3"></a>segmentation_lines <span class="ot">&lt;-</span> <span class="fu">st_read</span>(<span class="st">"Data_files/polygon_cuts.kml"</span>)</span>
<span id="cb8-4"><a href="#cb8-4"></a></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co"># Map the lines </span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="co"># mapview(segmentation_lines) </span></span>
<span id="cb8-7"><a href="#cb8-7"></a></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co"># Before we cut the polygons using the segmentation lines we mustrepair the geometry of the polygon and lines data for any errors </span></span>
<span id="cb8-9"><a href="#cb8-9"></a>polygon_valid <span class="ot">&lt;-</span> <span class="fu">st_make_valid</span>(polygon)</span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co"># mapview(polygon_valid)</span></span>
<span id="cb8-11"><a href="#cb8-11"></a></span>
<span id="cb8-12"><a href="#cb8-12"></a>lines_valid <span class="ot">&lt;-</span> <span class="fu">st_make_valid</span>(segmentation_lines)</span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co"># mapview(lines_valid)</span></span>
<span id="cb8-14"><a href="#cb8-14"></a></span>
<span id="cb8-15"><a href="#cb8-15"></a><span class="co"># Now cut the polygon using the segmentation lines </span></span>
<span id="cb8-16"><a href="#cb8-16"></a>cut_polygon <span class="ot">&lt;-</span> polygon_valid <span class="sc">%&gt;%</span> </span>
<span id="cb8-17"><a href="#cb8-17"></a>  lwgeom<span class="sc">::</span><span class="fu">st_split</span>(lines_valid) <span class="sc">%&gt;%</span> </span>
<span id="cb8-18"><a href="#cb8-18"></a>  <span class="fu">st_collection_extract</span>(<span class="st">"POLYGON"</span>)</span>
<span id="cb8-19"><a href="#cb8-19"></a></span>
<span id="cb8-20"><a href="#cb8-20"></a>p <span class="ot">&lt;-</span> <span class="fu">mapview</span>(cut_polygon)</span>
<span id="cb8-21"><a href="#cb8-21"></a></span>
<span id="cb8-22"><a href="#cb8-22"></a><span class="fu">mapshot</span>(p, <span class="at">file =</span> <span class="st">"Data_files/cut_polygon.png"</span>)</span>
<span id="cb8-23"><a href="#cb8-23"></a></span>
<span id="cb8-24"><a href="#cb8-24"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb8-25"><a href="#cb8-25"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb8-26"><a href="#cb8-26"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb8-27"><a href="#cb8-27"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb8-28"><a href="#cb8-28"></a>  p</span>
<span id="cb8-29"><a href="#cb8-29"></a>} <span class="cf">else</span> {</span>
<span id="cb8-30"><a href="#cb8-30"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb8-31"><a href="#cb8-31"></a>  p</span>
<span id="cb8-32"><a href="#cb8-32"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data_files/cut_polygon.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</div>
</section>
<section id="calculate-the-species-richness-in-each-of-the-new-polygons" class="level2">
<h2 class="anchored" data-anchor-id="calculate-the-species-richness-in-each-of-the-new-polygons">Calculate the species richness in each of the new polygons</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># Now we will calculate the number of species in each polygon segment.We are interested in polygons with IDs.</span></span>
<span id="cb9-2"><a href="#cb9-2"></a></span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="co"># The cut_polygon object is an 'sfc' object and we want to make it an 'sf' object</span></span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="fu">class</span>(cut_polygon) </span>
<span id="cb9-5"><a href="#cb9-5"></a></span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="co"># Create a new data frame</span></span>
<span id="cb9-7"><a href="#cb9-7"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(cut_polygon)) </span>
<span id="cb9-8"><a href="#cb9-8"></a></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co"># Add the sfc_POLYGON object as a geometry column</span></span>
<span id="cb9-10"><a href="#cb9-10"></a>df<span class="sc">$</span>geometry <span class="ot">&lt;-</span> cut_polygon </span>
<span id="cb9-11"><a href="#cb9-11"></a></span>
<span id="cb9-12"><a href="#cb9-12"></a><span class="co"># Convert the data frame to an sf object</span></span>
<span id="cb9-13"><a href="#cb9-13"></a>cut_polygon_sf <span class="ot">&lt;-</span> <span class="fu">st_as_sf</span>(df)  </span>
<span id="cb9-14"><a href="#cb9-14"></a></span>
<span id="cb9-15"><a href="#cb9-15"></a><span class="co"># Now an sf object with 72 polygons.</span></span>
<span id="cb9-16"><a href="#cb9-16"></a><span class="co"># str(cut_polygon_sf)</span></span>
<span id="cb9-17"><a href="#cb9-17"></a></span>
<span id="cb9-18"><a href="#cb9-18"></a><span class="fu">st_write</span>(cut_polygon_sf, <span class="st">"Data_files/cut_polygon_plot_sf.kml"</span>, <span class="at">append=</span> <span class="cn">FALSE</span>, <span class="at">driver =</span> <span class="st">"KML"</span>)</span>
<span id="cb9-19"><a href="#cb9-19"></a></span>
<span id="cb9-20"><a href="#cb9-20"></a><span class="co"># Now determine which polygon each point falls into </span></span>
<span id="cb9-21"><a href="#cb9-21"></a>result <span class="ot">&lt;-</span> <span class="fu">st_join</span>(df_species_jittered, cut_polygon_sf)</span>
<span id="cb9-22"><a href="#cb9-22"></a></span>
<span id="cb9-23"><a href="#cb9-23"></a><span class="co"># Convert the polygon ID to character </span></span>
<span id="cb9-24"><a href="#cb9-24"></a>result<span class="sc">$</span>id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(result<span class="sc">$</span>id) </span>
<span id="cb9-25"><a href="#cb9-25"></a></span>
<span id="cb9-26"><a href="#cb9-26"></a><span class="co"># Now we calculate the species diversity of each polygon </span></span>
<span id="cb9-27"><a href="#cb9-27"></a></span>
<span id="cb9-28"><a href="#cb9-28"></a>species_div <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb9-29"><a href="#cb9-29"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span></span>
<span id="cb9-30"><a href="#cb9-30"></a>  <span class="fu">summarise</span>(<span class="at">num_species =</span> <span class="fu">n_distinct</span>(speciesReorder))</span>
<span id="cb9-31"><a href="#cb9-31"></a></span>
<span id="cb9-32"><a href="#cb9-32"></a>species_div</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plot-the-centroids-of-each-polygon-with-species-richness-of-the-polygon-image-for-figure-5f" class="level2">
<h2 class="anchored" data-anchor-id="plot-the-centroids-of-each-polygon-with-species-richness-of-the-polygon-image-for-figure-5f">Plot the centroids of each polygon with species richness of the polygon (image for Figure 5f)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># To enable plotting we calculate the centroids of each polygon</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>average_geometry <span class="ot">&lt;-</span> <span class="fu">st_centroid</span>(species_div<span class="sc">$</span>geometry)</span>
<span id="cb10-3"><a href="#cb10-3"></a>species_div_centroids <span class="ot">&lt;-</span> <span class="fu">st_set_geometry</span>(species_div, average_geometry)</span>
<span id="cb10-4"><a href="#cb10-4"></a></span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="co"># mapview(species_div_centroids)</span></span>
<span id="cb10-6"><a href="#cb10-6"></a></span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="co"># We can select the centroids along the west side of the transect using this code if we want, default is to plot all</span></span>
<span id="cb10-8"><a href="#cb10-8"></a></span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="co"># id_values &lt;- c(14:23, 25:38, 2:7, 9)</span></span>
<span id="cb10-10"><a href="#cb10-10"></a><span class="co"># Species_div_centroids_subset &lt;- species_div_centroids[species_div_centroids$id %in% id_values, ]</span></span>
<span id="cb10-11"><a href="#cb10-11"></a></span>
<span id="cb10-12"><a href="#cb10-12"></a><span class="co"># Rename</span></span>
<span id="cb10-13"><a href="#cb10-13"></a>species_div_centroids_subset <span class="ot">&lt;-</span> species_div_centroids </span>
<span id="cb10-14"><a href="#cb10-14"></a></span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="co"># Removed the very large unsegmented polygon</span></span>
<span id="cb10-16"><a href="#cb10-16"></a>species_div_centroids_subset <span class="ot">&lt;-</span> species_div_centroids_subset[species_div_centroids_subset<span class="sc">$</span>num_species <span class="sc">&lt;</span> <span class="dv">7</span>, ]</span>
<span id="cb10-17"><a href="#cb10-17"></a></span>
<span id="cb10-18"><a href="#cb10-18"></a><span class="co"># Check the data</span></span>
<span id="cb10-19"><a href="#cb10-19"></a><span class="co"># mapview(species_div_centroids_subset)</span></span>
<span id="cb10-20"><a href="#cb10-20"></a></span>
<span id="cb10-21"><a href="#cb10-21"></a><span class="fu">mapviewOptions</span>(<span class="at">basemaps =</span> <span class="fu">c</span>(<span class="st">"Esri.WorldImagery"</span>),</span>
<span id="cb10-22"><a href="#cb10-22"></a>               <span class="at">raster.palette =</span> grey.colors,</span>
<span id="cb10-23"><a href="#cb10-23"></a>               <span class="at">vector.palette =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"red3"</span>)),</span>
<span id="cb10-24"><a href="#cb10-24"></a>               <span class="at">na.color =</span> <span class="st">"gray"</span>,</span>
<span id="cb10-25"><a href="#cb10-25"></a>               <span class="at">layers.control.pos =</span> <span class="st">"topright"</span>)</span>
<span id="cb10-26"><a href="#cb10-26"></a></span>
<span id="cb10-27"><a href="#cb10-27"></a>q <span class="ot">&lt;-</span> <span class="fu">mapview</span>(species_div_centroids_subset, <span class="at">zcol =</span> <span class="st">"num_species"</span>, <span class="at">layer.name =</span> <span class="st">"Richness"</span>, <span class="at">cex =</span> <span class="dv">8</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-28"><a href="#cb10-28"></a></span>
<span id="cb10-29"><a href="#cb10-29"></a><span class="fu">mapshot</span>(q, <span class="at">file =</span> <span class="st">"Data_files/Fig_5f.png"</span>)</span>
<span id="cb10-30"><a href="#cb10-30"></a></span>
<span id="cb10-31"><a href="#cb10-31"></a></span>
<span id="cb10-32"><a href="#cb10-32"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb10-33"><a href="#cb10-33"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb10-34"><a href="#cb10-34"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb10-35"><a href="#cb10-35"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb10-36"><a href="#cb10-36"></a>  q</span>
<span id="cb10-37"><a href="#cb10-37"></a>} <span class="cf">else</span> {</span>
<span id="cb10-38"><a href="#cb10-38"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb10-39"><a href="#cb10-39"></a>  q</span>
<span id="cb10-40"><a href="#cb10-40"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data_files/Fig_5f.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</div>
</section>
</section>
<section id="analysis-of-spinifex-data" class="level1">
<h1>Analysis of spinifex data</h1>
<section id="check-the-data-and-correct-outliers" class="level2">
<h2 class="anchored" data-anchor-id="check-the-data-and-correct-outliers">Check the data and correct outliers</h2>
<p>Here we are interested in analysing two variables: the % cover of live and dead material within the 10 m diameter plots.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># Ensure variables are numeric</span></span>
<span id="cb11-2"><a href="#cb11-2"></a>df_spinifex<span class="sc">$</span>live_cover<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df_spinifex<span class="sc">$</span>live_cover)</span>
<span id="cb11-3"><a href="#cb11-3"></a>df_spinifex<span class="sc">$</span>dead_cover<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df_spinifex<span class="sc">$</span>dead_cover)</span>
<span id="cb11-4"><a href="#cb11-4"></a></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="co"># Check the range of the data fields of interest. Percentage cover is the % cover within the 10 m diameter plots, impact score is based on the percentage of dead versus live material.</span></span>
<span id="cb11-6"><a href="#cb11-6"></a></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co"># Within range</span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="fu">summary</span>(df_spinifex<span class="sc">$</span>live_cover) </span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co"># Within range</span></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="fu">summary</span>(df_spinifex<span class="sc">$</span>dead_cover) </span>
<span id="cb11-11"><a href="#cb11-11"></a></span>
<span id="cb11-12"><a href="#cb11-12"></a><span class="co"># Generate an object with total live and dead cover in each frame </span></span>
<span id="cb11-13"><a href="#cb11-13"></a>frame_sum <span class="ot">&lt;-</span> df_spinifex <span class="sc">%&gt;%</span> </span>
<span id="cb11-14"><a href="#cb11-14"></a>  <span class="fu">group_by</span>(frame) <span class="sc">%&gt;%</span></span>
<span id="cb11-15"><a href="#cb11-15"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">c</span>(dead_cover, live_cover), sum, <span class="at">.names =</span> <span class="st">"{.col}"</span>))</span>
<span id="cb11-16"><a href="#cb11-16"></a></span>
<span id="cb11-17"><a href="#cb11-17"></a><span class="co"># Calculate the total spinifex cover variable </span></span>
<span id="cb11-18"><a href="#cb11-18"></a>frame_sum<span class="sc">$</span>total_cover <span class="ot">&lt;-</span> frame_sum<span class="sc">$</span>dead_cover<span class="sc">+</span> frame_sum<span class="sc">$</span>live_cover </span>
<span id="cb11-19"><a href="#cb11-19"></a></span>
<span id="cb11-20"><a href="#cb11-20"></a>frame_sum</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="map-the-total-cover-of-live-and-dead-cover-within-each-image-frame-images-for-figures-6c-and-6d" class="level2">
<h2 class="anchored" data-anchor-id="map-the-total-cover-of-live-and-dead-cover-within-each-image-frame-images-for-figures-6c-and-6d">Map the total cover of live and dead cover within each image frame (images for Figures 6c and 6d)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># Rename the object</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>data <span class="ot">&lt;-</span> frame_sum  </span>
<span id="cb12-3"><a href="#cb12-3"></a></span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="co"># Set the mapview options for live cover</span></span>
<span id="cb12-5"><a href="#cb12-5"></a><span class="fu">mapviewOptions</span>(<span class="at">basemaps =</span> <span class="fu">c</span>(<span class="st">"Esri.WorldImagery"</span>),</span>
<span id="cb12-6"><a href="#cb12-6"></a>               <span class="at">raster.palette =</span> grey.colors,</span>
<span id="cb12-7"><a href="#cb12-7"></a>               <span class="at">vector.palette =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"transparent"</span>, <span class="st">"green"</span>, <span class="st">'darkgreen'</span>)),</span>
<span id="cb12-8"><a href="#cb12-8"></a>               <span class="at">na.color =</span> <span class="st">"gray"</span>,</span>
<span id="cb12-9"><a href="#cb12-9"></a>               <span class="at">layers.control.pos =</span> <span class="st">"topright"</span>)</span>
<span id="cb12-10"><a href="#cb12-10"></a></span>
<span id="cb12-11"><a href="#cb12-11"></a><span class="co"># Plot of live cover </span></span>
<span id="cb12-12"><a href="#cb12-12"></a>p <span class="ot">&lt;-</span> <span class="fu">mapview</span>(data, <span class="at">zcol =</span> <span class="st">"live_cover"</span>, <span class="at">layer.name =</span> <span class="st">"Cover(%)"</span>, <span class="at">cex =</span> <span class="dv">10</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>,  <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-13"><a href="#cb12-13"></a></span>
<span id="cb12-14"><a href="#cb12-14"></a><span class="co">#p</span></span>
<span id="cb12-15"><a href="#cb12-15"></a></span>
<span id="cb12-16"><a href="#cb12-16"></a><span class="fu">mapshot</span>(p, <span class="at">file =</span> <span class="st">"Data_files/Fig_6c.png"</span>)</span>
<span id="cb12-17"><a href="#cb12-17"></a></span>
<span id="cb12-18"><a href="#cb12-18"></a><span class="co"># Set the mapview options for dead cover</span></span>
<span id="cb12-19"><a href="#cb12-19"></a><span class="fu">mapviewOptions</span>(<span class="at">basemaps =</span> <span class="fu">c</span>(<span class="st">"Esri.WorldImagery"</span>),</span>
<span id="cb12-20"><a href="#cb12-20"></a>               <span class="at">raster.palette =</span> grey.colors,</span>
<span id="cb12-21"><a href="#cb12-21"></a>               <span class="at">vector.palette =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"red"</span>, <span class="st">"darkred"</span>)),</span>
<span id="cb12-22"><a href="#cb12-22"></a>               <span class="at">na.color =</span> <span class="st">"gray"</span>,</span>
<span id="cb12-23"><a href="#cb12-23"></a>               <span class="at">layers.control.pos =</span> <span class="st">"topright"</span>)</span>
<span id="cb12-24"><a href="#cb12-24"></a></span>
<span id="cb12-25"><a href="#cb12-25"></a><span class="co"># Plot of dead cover</span></span>
<span id="cb12-26"><a href="#cb12-26"></a>q <span class="ot">&lt;-</span> <span class="fu">mapview</span>(data, <span class="at">zcol =</span> <span class="st">"dead_cover"</span>, <span class="at">layer.name =</span> <span class="st">"Cover(%)"</span>, <span class="at">cex =</span> <span class="dv">10</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-27"><a href="#cb12-27"></a></span>
<span id="cb12-28"><a href="#cb12-28"></a><span class="fu">mapshot</span>(q, <span class="at">file =</span> <span class="st">"Data_files/Fig_6d.png"</span>)</span>
<span id="cb12-29"><a href="#cb12-29"></a></span>
<span id="cb12-30"><a href="#cb12-30"></a><span class="co">#q</span></span>
<span id="cb12-31"><a href="#cb12-31"></a></span>
<span id="cb12-32"><a href="#cb12-32"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb12-33"><a href="#cb12-33"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb12-34"><a href="#cb12-34"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb12-35"><a href="#cb12-35"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb12-36"><a href="#cb12-36"></a>  p</span>
<span id="cb12-37"><a href="#cb12-37"></a>  </span>
<span id="cb12-38"><a href="#cb12-38"></a>  q</span>
<span id="cb12-39"><a href="#cb12-39"></a>} <span class="cf">else</span> {</span>
<span id="cb12-40"><a href="#cb12-40"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb12-41"><a href="#cb12-41"></a>  p</span>
<span id="cb12-42"><a href="#cb12-42"></a>  </span>
<span id="cb12-43"><a href="#cb12-43"></a>  q</span>
<span id="cb12-44"><a href="#cb12-44"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<p><img src="Data_files/Fig_6c.png" class="img-fluid quarto-figure quarto-figure-center"> <img src="Data_files/Fig_6d.png" class="img-fluid quarto-figure quarto-figure-center"></p>
</div>
</section>
<section id="map-the-cover-of-different-spinifex-size-classes-within-each-image-frame-image-for-figure-6e" class="level2">
<h2 class="anchored" data-anchor-id="map-the-cover-of-different-spinifex-size-classes-within-each-image-frame-image-for-figure-6e">Map the cover of different spinifex size classes within each image frame (image for Figure 6e)</h2>
</section>
<section id="re-name-spinifex-speciessize-data-for-size-class-analysis" class="level2">
<h2 class="anchored" data-anchor-id="re-name-spinifex-speciessize-data-for-size-class-analysis">Re-name spinifex species/size data for size class analysis</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'No Triodia Present'</span>] <span class="ot">&lt;-</span> <span class="st">'Absent'</span></span>
<span id="cb13-2"><a href="#cb13-2"></a></span>
<span id="cb13-3"><a href="#cb13-3"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia pungens  &gt;100cm old fragmentary'</span>] <span class="ot">&lt;-</span> <span class="st">'frag'</span></span>
<span id="cb13-4"><a href="#cb13-4"></a></span>
<span id="cb13-5"><a href="#cb13-5"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia sp. &gt;100cm old fragmentary'</span>] <span class="ot">&lt;-</span> <span class="st">'frag'</span></span>
<span id="cb13-6"><a href="#cb13-6"></a></span>
<span id="cb13-7"><a href="#cb13-7"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia pungens &lt; 50 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'&lt;50'</span></span>
<span id="cb13-8"><a href="#cb13-8"></a></span>
<span id="cb13-9"><a href="#cb13-9"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia sp. 50-&lt;100 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'50_100'</span></span>
<span id="cb13-10"><a href="#cb13-10"></a></span>
<span id="cb13-11"><a href="#cb13-11"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia sp. 100-&lt;200 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'100_200'</span></span>
<span id="cb13-12"><a href="#cb13-12"></a></span>
<span id="cb13-13"><a href="#cb13-13"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia pungens 200-&lt;400 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'200_400'</span></span>
<span id="cb13-14"><a href="#cb13-14"></a></span>
<span id="cb13-15"><a href="#cb13-15"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia sp. &lt; 50 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'&lt;50'</span></span>
<span id="cb13-16"><a href="#cb13-16"></a></span>
<span id="cb13-17"><a href="#cb13-17"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia pungens 50-&lt;100 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'50_100'</span></span>
<span id="cb13-18"><a href="#cb13-18"></a></span>
<span id="cb13-19"><a href="#cb13-19"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia pungens 100-&lt;200 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'100_200'</span></span>
<span id="cb13-20"><a href="#cb13-20"></a></span>
<span id="cb13-21"><a href="#cb13-21"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia schinzii 50-&lt;100 cm'</span>] <span class="ot">&lt;-</span> <span class="st">'50_100'</span></span>
<span id="cb13-22"><a href="#cb13-22"></a></span>
<span id="cb13-23"><a href="#cb13-23"></a>df_spinifex<span class="sc">$</span>species[df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">'Triodia pungens 400+ cm'</span>] <span class="ot">&lt;-</span> <span class="st">'&gt;400'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="calculate-and-map-the-cover-and-distribution-of-50cm-and-fragmentary-old-1m-size-classes-image-for-figure-6e" class="level2">
<h2 class="anchored" data-anchor-id="calculate-and-map-the-cover-and-distribution-of-50cm-and-fragmentary-old-1m-size-classes-image-for-figure-6e">Calculate and map the cover and distribution of &lt; 50cm and fragmentary (old &gt; 1m) size classes (image for Figure 6e)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="co"># Subset spinifex data for only &lt; 50 cm size class</span></span>
<span id="cb14-2"><a href="#cb14-2"></a>Less50cm <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_spinifex, df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">"&lt;50"</span>)</span>
<span id="cb14-3"><a href="#cb14-3"></a> </span>
<span id="cb14-4"><a href="#cb14-4"></a><span class="co"># Subset spinifex data for only old fragmentary size class</span></span>
<span id="cb14-5"><a href="#cb14-5"></a>frag <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_spinifex, df_spinifex<span class="sc">$</span>species <span class="sc">==</span> <span class="st">"frag"</span>)</span>
<span id="cb14-6"><a href="#cb14-6"></a></span>
<span id="cb14-7"><a href="#cb14-7"></a><span class="co"># Jitter the data so that they can be seen more easily in the map</span></span>
<span id="cb14-8"><a href="#cb14-8"></a>jittered_frag <span class="ot">&lt;-</span> <span class="fu">st_jitter</span>(frag, <span class="at">amount =</span> <span class="fl">0.0001</span>)  </span>
<span id="cb14-9"><a href="#cb14-9"></a>jittered_Less50cm <span class="ot">&lt;-</span> <span class="fu">st_jitter</span>(Less50cm, <span class="at">amount =</span> <span class="fl">0.0001</span>)</span>
<span id="cb14-10"><a href="#cb14-10"></a></span>
<span id="cb14-11"><a href="#cb14-11"></a><span class="co"># Map the variables together - Figure 6e</span></span>
<span id="cb14-12"><a href="#cb14-12"></a>p <span class="ot">&lt;-</span> <span class="fu">mapview</span>(jittered_frag, <span class="at">zcol =</span> <span class="st">"percentage_cover"</span>, <span class="at">layer.name =</span> <span class="st">"Frag_Cover(%)"</span>, <span class="at">cex =</span> <span class="dv">8</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">alpha.regions =</span> <span class="fl">0.75</span>, <span class="at">col.regions =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"yellow"</span>, <span class="st">"orange"</span>, <span class="st">'orange4'</span>)), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)  <span class="sc">+</span> <span class="fu">mapview</span>(jittered_Less50cm, <span class="at">zcol =</span> <span class="st">"percentage_cover"</span>, <span class="at">cex =</span> <span class="dv">8</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>, <span class="at">col.regions =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"lightblue"</span>, <span class="st">"blue"</span>, <span class="st">'darkblue'</span>)),<span class="at">layer.name =</span> <span class="st">"L50_Cover(%)"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-13"><a href="#cb14-13"></a></span>
<span id="cb14-14"><a href="#cb14-14"></a><span class="fu">mapshot</span>(p, <span class="at">file =</span> <span class="st">"Data_files/Fig_6e.png"</span>)</span>
<span id="cb14-15"><a href="#cb14-15"></a></span>
<span id="cb14-16"><a href="#cb14-16"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb14-17"><a href="#cb14-17"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb14-18"><a href="#cb14-18"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb14-19"><a href="#cb14-19"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb14-20"><a href="#cb14-20"></a>  p</span>
<span id="cb14-21"><a href="#cb14-21"></a>} <span class="cf">else</span> {</span>
<span id="cb14-22"><a href="#cb14-22"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb14-23"><a href="#cb14-23"></a>  p</span>
<span id="cb14-24"><a href="#cb14-24"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data_files/Fig_6e.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</div>
</section>
<section id="create-boxplot-of-size-class-vs-pre-drought-and-post-drought-percentage-cover-with-kruskal-wallis-test-image-for-figure-6f" class="level2">
<h2 class="anchored" data-anchor-id="create-boxplot-of-size-class-vs-pre-drought-and-post-drought-percentage-cover-with-kruskal-wallis-test-image-for-figure-6f">Create boxplot of size class vs pre-drought and post-drought percentage cover with kruskal-wallis test (image for Figure 6f)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>df_spinifex_full_data <span class="ot">&lt;-</span> df_spinifex</span>
<span id="cb15-2"><a href="#cb15-2"></a></span>
<span id="cb15-3"><a href="#cb15-3"></a>df_spinifex <span class="ot">&lt;-</span> df_spinifex <span class="sc">%&gt;%</span></span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">filter</span>(percentage_cover <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb15-5"><a href="#cb15-5"></a></span>
<span id="cb15-6"><a href="#cb15-6"></a>df_spinifex <span class="ot">&lt;-</span> df_spinifex[df_spinifex<span class="sc">$</span>species <span class="sc">!=</span> <span class="st">'Absent'</span>, ]</span>
<span id="cb15-7"><a href="#cb15-7"></a>df_spinifex <span class="ot">&lt;-</span> df_spinifex[df_spinifex<span class="sc">$</span>species <span class="sc">!=</span> <span class="st">'NA'</span>, ]</span>
<span id="cb15-8"><a href="#cb15-8"></a></span>
<span id="cb15-9"><a href="#cb15-9"></a>df_spinifex<span class="sc">$</span>speciesReorder <span class="ot">&lt;-</span> <span class="fu">factor</span>(df_spinifex<span class="sc">$</span>species, <span class="fu">c</span>(<span class="st">"&lt;50"</span>, <span class="st">"50_100"</span>, <span class="st">"100_200"</span>, <span class="st">"200_400"</span>, <span class="st">"&gt;400"</span>, <span class="st">"frag"</span>))</span>
<span id="cb15-10"><a href="#cb15-10"></a></span>
<span id="cb15-11"><a href="#cb15-11"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_spinifex, <span class="fu">aes</span>(<span class="at">x =</span> speciesReorder, <span class="at">y =</span> percentage_cover)) <span class="sc">+</span></span>
<span id="cb15-12"><a href="#cb15-12"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-13"><a href="#cb15-13"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb15-14"><a href="#cb15-14"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb15-15"><a href="#cb15-15"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>))</span>
<span id="cb15-16"><a href="#cb15-16"></a></span>
<span id="cb15-17"><a href="#cb15-17"></a>p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb15-18"><a href="#cb15-18"></a>  <span class="fu">xlab</span>(<span class="st">"Size class (cm)"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Pre-drought cover"</span>) <span class="sc">+</span></span>
<span id="cb15-19"><a href="#cb15-19"></a>  <span class="fu">theme</span>(<span class="at">axis.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>), <span class="at">axis.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">14</span>,<span class="at">face=</span><span class="st">"bold"</span>))</span>
<span id="cb15-20"><a href="#cb15-20"></a></span>
<span id="cb15-21"><a href="#cb15-21"></a><span class="fu">ggsave</span>(<span class="st">"Data_files/Fig_6f.png"</span>, <span class="at">width =</span> <span class="fl">6.81</span>, <span class="at">height =</span> <span class="fl">6.45</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb15-22"><a href="#cb15-22"></a></span>
<span id="cb15-23"><a href="#cb15-23"></a><span class="co"># Test for nonparametric relationship between percentage cover and size class</span></span>
<span id="cb15-24"><a href="#cb15-24"></a></span>
<span id="cb15-25"><a href="#cb15-25"></a><span class="fu">kruskal.test</span>(percentage_cover <span class="sc">~</span> speciesReorder, <span class="at">data =</span> df_spinifex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><img src="Data_files/Fig_6f.png" class="img-fluid"></p>
</section>
</section>
<section id="analysis-of-desert-oak-crown-health-data" class="level1">
<h1>Analysis of desert oak crown health data</h1>
<section id="import-the-lookup-csv-containing-size-classes-and-match-to-size-class-data-in-the-df_oak-object" class="level2">
<h2 class="anchored" data-anchor-id="import-the-lookup-csv-containing-size-classes-and-match-to-size-class-data-in-the-df_oak-object">Import the lookup csv containing size classes and match to size class data in the df_oak object</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># The original df_oak data have been imported in code above </span></span>
<span id="cb16-2"><a href="#cb16-2"></a></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="co"># Import the match table</span></span>
<span id="cb16-4"><a href="#cb16-4"></a>df_oak_class <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"Data_files/Oak_sizes_reclass.csv"</span>)</span>
<span id="cb16-5"><a href="#cb16-5"></a></span>
<span id="cb16-6"><a href="#cb16-6"></a><span class="co"># We must create new variables to allow the match </span></span>
<span id="cb16-7"><a href="#cb16-7"></a>df_oak<span class="sc">$</span>Taxon <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb16-8"><a href="#cb16-8"></a>df_oak<span class="sc">$</span>Size_class <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb16-9"><a href="#cb16-9"></a>df_oak<span class="sc">$</span>Burnt <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb16-10"><a href="#cb16-10"></a>df_oak<span class="sc">$</span>Count <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb16-11"><a href="#cb16-11"></a></span>
<span id="cb16-12"><a href="#cb16-12"></a><span class="co"># We must remove whitespace and other incompatibilities</span></span>
<span id="cb16-13"><a href="#cb16-13"></a></span>
<span id="cb16-14"><a href="#cb16-14"></a>df_oak<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(df_oak<span class="sc">$</span>species, <span class="st">"&lt;.*?&gt;"</span>, <span class="st">""</span>)</span>
<span id="cb16-15"><a href="#cb16-15"></a>df_oak_class<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(df_oak_class<span class="sc">$</span>species, <span class="st">"&lt;.*?&gt;"</span>, <span class="st">""</span>)</span>
<span id="cb16-16"><a href="#cb16-16"></a></span>
<span id="cb16-17"><a href="#cb16-17"></a>df_oak<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">str_to_lower</span>(df_oak<span class="sc">$</span>species)</span>
<span id="cb16-18"><a href="#cb16-18"></a>df_oak<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(df_oak<span class="sc">$</span>species, <span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">""</span>)</span>
<span id="cb16-19"><a href="#cb16-19"></a></span>
<span id="cb16-20"><a href="#cb16-20"></a>df_oak_class<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">str_to_lower</span>(df_oak_class<span class="sc">$</span>species)</span>
<span id="cb16-21"><a href="#cb16-21"></a>df_oak_class<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(df_oak_class<span class="sc">$</span>species, <span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="st">""</span>)</span>
<span id="cb16-22"><a href="#cb16-22"></a></span>
<span id="cb16-23"><a href="#cb16-23"></a><span class="co"># Now merge Taxon, Burnt, Size_class, Species</span></span>
<span id="cb16-24"><a href="#cb16-24"></a></span>
<span id="cb16-25"><a href="#cb16-25"></a>df_oak<span class="sc">$</span>Taxon <span class="ot">&lt;-</span> df_oak_class<span class="sc">$</span>Taxon[<span class="fu">match</span>(df_oak<span class="sc">$</span>species, df_oak_class<span class="sc">$</span>species)]</span>
<span id="cb16-26"><a href="#cb16-26"></a></span>
<span id="cb16-27"><a href="#cb16-27"></a>df_oak<span class="sc">$</span>Burnt <span class="ot">&lt;-</span> df_oak_class<span class="sc">$</span>Burnt[<span class="fu">match</span>(df_oak<span class="sc">$</span>species, df_oak_class<span class="sc">$</span>species)]</span>
<span id="cb16-28"><a href="#cb16-28"></a></span>
<span id="cb16-29"><a href="#cb16-29"></a>df_oak<span class="sc">$</span>Count <span class="ot">&lt;-</span> df_oak_class<span class="sc">$</span>Count[<span class="fu">match</span>(df_oak<span class="sc">$</span>species, df_oak_class<span class="sc">$</span>species)]</span>
<span id="cb16-30"><a href="#cb16-30"></a></span>
<span id="cb16-31"><a href="#cb16-31"></a>df_oak<span class="sc">$</span>Size_class <span class="ot">&lt;-</span> df_oak_class<span class="sc">$</span>Size_class[<span class="fu">match</span>(df_oak<span class="sc">$</span>species, df_oak_class<span class="sc">$</span>species)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="calculate-crown-health-score-and-compare-crown-health-across-size-classes-create-image-for-figure-7f" class="level2">
<h2 class="anchored" data-anchor-id="calculate-crown-health-score-and-compare-crown-health-across-size-classes-create-image-for-figure-7f">Calculate crown health score and compare crown health across size classes (create image for Figure 7f)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="co"># Ensure that the variables are numeric </span></span>
<span id="cb17-2"><a href="#cb17-2"></a>df_oak<span class="sc">$</span>dd1 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df_oak<span class="sc">$</span>dd1) </span>
<span id="cb17-3"><a href="#cb17-3"></a>df_oak<span class="sc">$</span>dd2 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df_oak<span class="sc">$</span>dd2)</span>
<span id="cb17-4"><a href="#cb17-4"></a>df_oak<span class="sc">$</span>dd3 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df_oak<span class="sc">$</span>dd3)</span>
<span id="cb17-5"><a href="#cb17-5"></a></span>
<span id="cb17-6"><a href="#cb17-6"></a><span class="co"># Calculate the crown score (HC = AC  CE  CD in paper)</span></span>
<span id="cb17-7"><a href="#cb17-7"></a>df_oak<span class="sc">$</span>Crown_score <span class="ot">&lt;-</span> df_oak<span class="sc">$</span>dd1<span class="sc">/</span><span class="dv">100</span> <span class="sc">*</span> df_oak<span class="sc">$</span>dd2<span class="sc">/</span><span class="dv">100</span> <span class="sc">*</span> df_oak<span class="sc">$</span>dd3<span class="sc">/</span><span class="dv">100</span></span>
<span id="cb17-8"><a href="#cb17-8"></a></span>
<span id="cb17-9"><a href="#cb17-9"></a><span class="co"># Reorder the classes in order of size </span></span>
<span id="cb17-10"><a href="#cb17-10"></a>df_oak<span class="sc">$</span>Size_class_Reorder <span class="ot">&lt;-</span> <span class="fu">factor</span>(df_oak<span class="sc">$</span>Size_class, <span class="fu">c</span>(<span class="st">"Seedling"</span>, <span class="st">"Juvenile"</span>, <span class="st">"Early_Pole"</span>, <span class="st">"Late_Pole"</span>, <span class="st">"Early_Mature"</span>, <span class="st">"Mature"</span>))</span>
<span id="cb17-11"><a href="#cb17-11"></a></span>
<span id="cb17-12"><a href="#cb17-12"></a><span class="co"># Boxplot of size class and burnt vs crown score</span></span>
<span id="cb17-13"><a href="#cb17-13"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_oak, <span class="fu">aes</span>(<span class="at">x =</span> Size_class_Reorder, <span class="at">y =</span> Crown_score,  <span class="at">z =</span> Burnt, <span class="at">fill =</span> Burnt, <span class="at">color =</span> Burnt )) <span class="sc">+</span></span>
<span id="cb17-14"><a href="#cb17-14"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-15"><a href="#cb17-15"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb17-16"><a href="#cb17-16"></a>      <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"green4"</span>,<span class="st">"brown3"</span>)) <span class="sc">+</span></span>
<span id="cb17-17"><a href="#cb17-17"></a>      <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgreen"</span>,<span class="st">"brown4"</span>)) <span class="sc">+</span></span>
<span id="cb17-18"><a href="#cb17-18"></a>      <span class="fu">theme_classic</span>()  <span class="sc">+</span></span>
<span id="cb17-19"><a href="#cb17-19"></a>      <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">size =</span> <span class="dv">2</span>))</span>
<span id="cb17-20"><a href="#cb17-20"></a></span>
<span id="cb17-21"><a href="#cb17-21"></a>p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb17-22"><a href="#cb17-22"></a>  <span class="fu">xlab</span>(<span class="st">"Size class"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Crown score"</span>) <span class="sc">+</span></span>
<span id="cb17-23"><a href="#cb17-23"></a>  <span class="fu">theme</span>(<span class="at">axis.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>), <span class="at">axis.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">14</span>,<span class="at">face=</span><span class="st">"bold"</span>))</span>
<span id="cb17-24"><a href="#cb17-24"></a></span>
<span id="cb17-25"><a href="#cb17-25"></a><span class="fu">ggsave</span>(<span class="st">"Data_files/Fig_7f.png"</span>, <span class="at">width =</span> <span class="fl">7.25</span>, <span class="at">height =</span> <span class="fl">6.45</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb17-26"><a href="#cb17-26"></a></span>
<span id="cb17-27"><a href="#cb17-27"></a><span class="co"># Test for nonparametric relationship between crown score and burnt + size class </span></span>
<span id="cb17-28"><a href="#cb17-28"></a><span class="fu">scheirerRayHare</span>(Crown_score <span class="sc">~</span> Burnt <span class="sc">+</span> Size_class_Reorder, <span class="at">data =</span> df_oak)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><img src="Data_files/Fig_7f.png" class="img-fluid"></p>
</section>
<section id="map-burnt-and-unburnt-tree-crowns-image-for-figure-7e" class="level2">
<h2 class="anchored" data-anchor-id="map-burnt-and-unburnt-tree-crowns-image-for-figure-7e">Map burnt and unburnt tree crowns (image for Figure 7e)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># Map burnt and unburnt areas </span></span>
<span id="cb18-2"><a href="#cb18-2"></a>p <span class="ot">&lt;-</span> <span class="fu">mapview</span>(df_oak, <span class="at">zcol =</span> <span class="st">"Burnt"</span>, <span class="at">cex =</span> <span class="dv">8</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>, <span class="at">layer.name =</span> <span class="st">"Burnt"</span>, <span class="at">col.regions =</span> <span class="fu">c</span>(<span class="st">"green2"</span>, <span class="st">"red2"</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-3"><a href="#cb18-3"></a></span>
<span id="cb18-4"><a href="#cb18-4"></a><span class="fu">mapshot</span>(p, <span class="at">file =</span> <span class="st">"Data_files/Fig_7e.png"</span>)</span>
<span id="cb18-5"><a href="#cb18-5"></a></span>
<span id="cb18-6"><a href="#cb18-6"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb18-8"><a href="#cb18-8"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb18-9"><a href="#cb18-9"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb18-10"><a href="#cb18-10"></a>  p</span>
<span id="cb18-11"><a href="#cb18-11"></a>} <span class="cf">else</span> {</span>
<span id="cb18-12"><a href="#cb18-12"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb18-13"><a href="#cb18-13"></a>  p</span>
<span id="cb18-14"><a href="#cb18-14"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data_files/Fig_7e.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</div>
</section>
<section id="map-the-distribution-of-tree-size-classes-image-for-figure-7c" class="level2">
<h2 class="anchored" data-anchor-id="map-the-distribution-of-tree-size-classes-image-for-figure-7c">Map the distribution of tree size classes (image for Figure 7c)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="co"># Map size classes </span></span>
<span id="cb19-2"><a href="#cb19-2"></a>p <span class="ot">&lt;-</span> <span class="fu">mapview</span>(df_oak, <span class="at">zcol =</span> <span class="st">"Size_class_Reorder"</span>, <span class="at">cex =</span> <span class="dv">8</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>, <span class="at">layer.name =</span> <span class="st">"Size_class"</span>, <span class="at">col.regions =</span> <span class="fu">c</span>(<span class="st">"#389f0a"</span>, <span class="st">"#389f0a"</span>,<span class="st">"yellow"</span>, <span class="st">"yellow"</span>, <span class="st">"red4"</span>, <span class="st">"red4"</span>),<span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-3"><a href="#cb19-3"></a></span>
<span id="cb19-4"><a href="#cb19-4"></a><span class="fu">mapshot</span>(p, <span class="at">file =</span> <span class="st">"Data_files/Fig_7c.png"</span>)</span>
<span id="cb19-5"><a href="#cb19-5"></a></span>
<span id="cb19-6"><a href="#cb19-6"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb19-7"><a href="#cb19-7"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb19-8"><a href="#cb19-8"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb19-9"><a href="#cb19-9"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb19-10"><a href="#cb19-10"></a>  p</span>
<span id="cb19-11"><a href="#cb19-11"></a>} <span class="cf">else</span> {</span>
<span id="cb19-12"><a href="#cb19-12"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb19-13"><a href="#cb19-13"></a>  p</span>
<span id="cb19-14"><a href="#cb19-14"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data_files/Fig_7c.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</div>
</section>
<section id="plot-the-crown-health-of-mature-and-early-mature-size-classes-figure-7d" class="level2">
<h2 class="anchored" data-anchor-id="plot-the-crown-health-of-mature-and-early-mature-size-classes-figure-7d">Plot the crown health of mature and early mature size classes (Figure 7d)</h2>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="co"># Select only mature and early mature size classes </span></span>
<span id="cb20-2"><a href="#cb20-2"></a></span>
<span id="cb20-3"><a href="#cb20-3"></a>Mature <span class="ot">&lt;-</span> <span class="fu">subset</span>(df_oak, df_oak<span class="sc">$</span>Size_class <span class="sc">==</span> <span class="st">"Mature"</span> <span class="sc">|</span> df_oak<span class="sc">$</span>Size_class <span class="sc">==</span> <span class="st">"Early_mature"</span>)</span>
<span id="cb20-4"><a href="#cb20-4"></a></span>
<span id="cb20-5"><a href="#cb20-5"></a><span class="fu">mapviewOptions</span>(<span class="at">basemaps =</span> <span class="fu">c</span>(<span class="st">"Esri.WorldImagery"</span>),</span>
<span id="cb20-6"><a href="#cb20-6"></a>               <span class="at">raster.palette =</span> grey.colors,</span>
<span id="cb20-7"><a href="#cb20-7"></a>               <span class="at">vector.palette =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"red"</span>, <span class="st">"yellow"</span>, <span class="st">"green2"</span>)),</span>
<span id="cb20-8"><a href="#cb20-8"></a>               <span class="at">na.color =</span> <span class="st">"gray"</span>,</span>
<span id="cb20-9"><a href="#cb20-9"></a>               <span class="at">layers.control.pos =</span> <span class="st">"topright"</span>)</span>
<span id="cb20-10"><a href="#cb20-10"></a></span>
<span id="cb20-11"><a href="#cb20-11"></a>p <span class="ot">&lt;-</span> <span class="fu">mapview</span>(df_oak, <span class="at">zcol =</span> <span class="st">"Crown_score"</span>, <span class="at">layer.name =</span> <span class="st">"Crown_score"</span>, <span class="at">cex =</span> <span class="dv">8</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">alpha.regions =</span> <span class="fl">0.9</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-12"><a href="#cb20-12"></a></span>
<span id="cb20-13"><a href="#cb20-13"></a><span class="fu">mapshot</span>(p, <span class="at">file =</span> <span class="st">"Data_files/Fig_7d.png"</span>)</span>
<span id="cb20-14"><a href="#cb20-14"></a></span>
<span id="cb20-15"><a href="#cb20-15"></a><span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_latex_output</span>()) {</span>
<span id="cb20-16"><a href="#cb20-16"></a>  <span class="co"># This part runs only for PDF output</span></span>
<span id="cb20-17"><a href="#cb20-17"></a>} <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="sc">::</span><span class="fu">is_html_output</span>()) {</span>
<span id="cb20-18"><a href="#cb20-18"></a>  <span class="co"># This part runs only for HTML output</span></span>
<span id="cb20-19"><a href="#cb20-19"></a>  p</span>
<span id="cb20-20"><a href="#cb20-20"></a>} <span class="cf">else</span> {</span>
<span id="cb20-21"><a href="#cb20-21"></a>  <span class="co"># This part runs if neither PDF nor HTML is the target format</span></span>
<span id="cb20-22"><a href="#cb20-22"></a>  p</span>
<span id="cb20-23"><a href="#cb20-23"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="pdf">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data_files/Fig_7d.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>